lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3053,15 +3103,16 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
   // Set the checked differentiation parameter indices in the attribute.
   attr->setParameterIndices(checkedWrtParamIndices);
 
-  // Check that original function's result type conforms to `Differentiable`.
-  if (whereClauseGenEnv) {
-    auto originalResultInterfaceType = !originalResultTy->hasTypeParameter()
-        ? originalResultTy->mapTypeOutOfContext()
-        : originalResultTy;
+  originalResultTy = originalResultTy->hasTypeParameter()
```

Comment:
For the same reasons, this expression will never change originalResultTy.

Reply:
Done in https://github.com/apple/swift/commit/38de6947fc62bc1c734d6f5f6f3359f3b5c93b13.